script/create_changelog

```diff
@@ -0,0 +1,250 @@
+	echo ""
+	echo -e "- \033[1mEMAIL\033[0m"
+	echo -e "   This is your email id, configured through \033[3mgit config --global user.email\033[0m, to sign the commit"
+	echo "   This script doesn't print your EMAIL on output"
+	echo ""
+	echo -e "\033[1mHow this script works?\033[0m"
+	echo -e "- \033[3mGenerating changelog\033[0m"
+	echo -e "   Script will use files from directory \033[2mchangelogs/unreleased\033[0m to create a changelog"
+	echo -e "   It will create a new directory, named \033[4mtag-name\033[0m, under directory \033[2mchangelogs/released\033[0m"
+	echo -e "   and move all the files from directory \033[2mchangelogs/unreleased\033[0m to new directory."
+	echo "   Once all the files are moved to new directory, it will create a temporary file "
+	echo -e "   in \033[2m/tmp/\033[0m directory and copy the content of all files, under new directory to a"
+	echo "   temporary file."
+	echo -e "   If tag is release tag, like \033[2mv1.1.0\033[0m, and there are existing directory for rc release,"
+	echo -e "   like \033[2mv1.1.0-RC1\033[0m, then script will copy the contents of all files, under this rc release "
+	echo "   directory, to a temporary file."
+	echo "   After this, script will add the content of temporary file to root changelog file "
+	echo "   and delete the temporary file"
+	echo ""
+	echo -e "- \033[3mCreating a branch\033[0m"
+	echo -e "  Script creates a new branch named \033[2mchangelog-timestamp\033[0m, like \033[2mchangelog-20200512164456\033[0m"
+	echo ""
+	echo -e "- \033[3mCreating a commit\033[0m"
+	echo -e "   Script will create commit with following files using your \033[1mUSERNAME\033[0m and \033[1mEMAIL\033[0m"
+	echo -e "   - \033[2mrepo-name/CHANGELOG.md\033[0m (\033[3mmodified\033[0m)"
+	echo -e "   - \033[2mrepo-name/changelogs/unreleased/*\033[0m (\033[3mdeleted\033[0m)"
+	echo -e "   - \033[2mrepo-name/changelogs/released/tag-name\033[0m (\033[3mnew directory\033[0m)"
+	echo ""
+	echo -e "- \033[3mPushing branch to repo\033[0m"
+	echo -e "   To push the changes to github, script will add new remote, named \033[2mupstream\033[0m, "
+	echo -e "   using your \033[1mUSERNAME\033[0m and \033[1mTOKEN\033[0m"
+	echo ""
+	echo -e "- \033[3mCreating a pull request\033[0m"
+	echo -e "   Script uses github API to create a pull request, using your \033[1mUSERNAME\033[0m and \033[1mTOKEN\033[0m."
+	echo "   Refer https://developer.github.com/v3/pulls/#create-a-pull-request for API"
+	exit 1
+fi
+
+if [ -z $USERNAME ] || [ -z $TOKEN ] || [ -z $EMAIL ]; then
+	echo "No crendentials provided for github"
+	exit 1
+fi
+
+BRANCH=changelog-$(date +'%Y%m%d%H%M%S')
+GIT_URL="https://api.github.com/repos/${1}/pulls"
+GIT_REPO=$1
+TAG=$2
+GIT_PRODUCTION_BRANCH=$3
+CHANGELOG_DIR=changelogs
+UNRELEASED_CHANGELOG_DIR=changelogs/unreleased
+RELEASED_CHANGELOG_DIR=changelogs/released
+ROOT_CHANGELOG=CHANGELOG.md
```

Comment:
question: @mynktl  This [release file](https://github.com/openebs/velero-plugin/blob/master/RELEASE.md) mentions about a `velero-plugin/CHANGELOG-{TAG}.md` file . Are we actually using it?

Reply:
No. I need opinion from other repo maintainers on that. Since we are following monthly release and two RC build, this will create lot of files on root directory within a year. Each month we will be adding 3 new files to root folder.